packages/fiber/src/core/events.ts

```diff
@@ -118,8 +118,12 @@ export function createEvents(store: UseStore<RootState>) {
     const { raycaster, mouse, camera, size } = state
     // https://github.com/pmndrs/react-three-fiber/pull/782
     // Events trigger outside of canvas when moved
-    const { offsetX, offsetY } = raycaster.computeOffsets?.(event, state) ?? event
-    const { width, height } = size
+    const customOffsets = raycaster.computeOffsets?.(event, state)
+    const offsetX = customOffsets?.offsetX ?? event.offsetX
+    const offsetY = customOffsets?.offsetY ?? event.offsetY
+    const width = customOffsets?.width ?? size.width
+    const height = customOffsets?.height ?? size.height
```

Comment:
How are you calculating width and height here? Should R3F respect the gl viewport or scissor regions?

Reply:
R3F should respect the size of the scissor regions. 

In `@react-three/scissor` the width and height pulled form the `BoundingClientRect` of the Scissor region.

Reply:
I see. This looks good as a workaround, but I'll want to investigate smarter defaults in v8.

Reply:
I agree, Happy to explore alternatives